File: Sources/Views/Settings/Panes/GeminiSettingsPane.swift
// MARK: - GeminiSettingsPane
// Settings pane for the Google Gemini CLI AI coding assistant.
// macOS 14+, Swift 5.10

import SwiftUI
import AppKit

// MARK: - GeminiSettingsPane

/// Settings pane for Google Gemini CLI.
///
/// Shows `~/.gemini/settings.json` with Finder + editor actions.
/// When the file does not exist yet a "Create config" button is offered.
struct GeminiSettingsPane: View {

    // MARK: ViewModel (lazy init)

    @State private var vmBox = LazyStateObject<GeminiSettingsPaneViewModel>()
    private var viewModel: GeminiSettingsPaneViewModel {
        vmBox.resolve { GeminiSettingsPaneViewModel() }
    }

    // MARK: State

    @State private var showEditor = false

    // MARK: - Body

    var body: some View {
        let model = viewModel
        ScrollView(.vertical, showsIndicators: true) {
            VStack(alignment: .leading, spacing: DSSpacing.xl) {
                Text("Gemini")
                    .font(DSFont.settingsTitle)
                    .foregroundStyle(DSColor.textPrimary)

                Divider().background(DSColor.borderDefault)

                settingsFileSection(model: model)

                authInfoRow
            }
            .padding(DSSpacing.xl)
            .frame(maxWidth: .infinity, alignment: .topLeading)
        }
        .frame(maxWidth: .infinity, maxHeight: .infinity)
        .onAppear { viewModel.checkConfigExists() }
        .sheet(isPresented: $showEditor, onDismiss: { viewModel.checkConfigExists() }, content: {
            TextFileEditorSheet(
                fileURL: GeminiSettingsPaneViewModel.settingsURL,
                displayTitle: "settings.json",
                defaultContent: defaultSettings
            )
        })
    }

    // MARK: - Settings File Section

    private func settingsFileSection(model: GeminiSettingsPaneViewModel) -> some View {
        VStack(alignment: .leading, spacing: DSSpacing.sm) {
            Text("Config")
                .font(DSFont.buttonLabel)
                .foregroundStyle(DSColor.textSecondary)

            SettingsConfigFileRow(
                displayPath: model.displayPath,
                configExists: model.configExists,
                onReveal: {
                    NSWorkspace.shared.activateFileViewerSelecting([GeminiSettingsPaneViewModel.settingsURL])
                },
                onEdit: { showEditor = true }
            )

            if model.configExists {
                settingsReferenceRow
            }
        }
    }

    // MARK: - Settings Reference

    private var settingsReferenceRow: some View {
        VStack(alignment: .leading, spacing: DSSpacing.xs) {
            Text("Available settings: theme, sandbox, checkpointing, preferredEditor,")
                .font(DSFont.sidebarItemSmall)
                .foregroundStyle(DSColor.textMuted)
            Text("contextWindowCompression, telemetry, coreTools, mcpServers, extensions.")
                .font(DSFont.sidebarItemSmall)
                .foregroundStyle(DSColor.textMuted)
        }
    }

    // MARK: - Auth Info Row

    private var authInfoRow: some View {
        SettingsAuthInfoRow(
            hint: "Set the API key via an environment variable:",
            envVar: "export GEMINI_API_KEY=your-key"
        )
    }

    // MARK: - Default Settings

    private let defaultSettings = """
        {
          "theme": "Default",
          "checkpointing": false,
          "contextWindowCompression": {
            "enabled": true,
            "threshold": 0.7
          },
          "telemetry": false
        }
        """
}

File: desktopApp/src/main/kotlin/studio/vibe/desktop/ui/settings/GeminiSettingsPane.kt
package studio.vibe.desktop.ui.settings

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.LaunchedEffect
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Modifier
import studio.vibe.desktop.DesktopServiceContainer
import studio.vibe.desktop.ui.theme.DSColor
import studio.vibe.desktop.ui.theme.LocalDSColors
import studio.vibe.desktop.ui.theme.DSFont
import studio.vibe.desktop.ui.theme.DSSpacing
import studio.vibe.shared.model.AIAssistant
import java.io.File

/**
 * Settings pane for Google Gemini CLI.
 *
 * Mirrors Swift GeminiSettingsPane.swift.
 *
 * Shows ~/.gemini/settings.json with existence check
 * and API key environment variable instructions.
 */
@Composable
fun GeminiSettingsPane(
    container: DesktopServiceContainer,
    modifier: Modifier = Modifier,
) {
    val homeDir = System.getProperty("user.home") ?: ""
    val settingsPath = remember { "$homeDir/.gemini/settings.json" }
    var configExists by remember { mutableStateOf(false) }

    LaunchedEffect(Unit) {
        configExists = File(settingsPath).exists()
    }

    LlmBasicPane(
        assistant = AIAssistant.GEMINI,
        modifier = modifier,
        extraContent = {
            // -- Config file ---------------------------------------------

            Column(
                modifier = Modifier.fillMaxWidth(),
                verticalArrangement = Arrangement.spacedBy(DSSpacing.sm),
            ) {
                Text(
                    text = "Config",
                    style = DSFont.buttonLabel,
                    color = LocalDSColors.current.textSecondary,
                )

                SettingsCard {
                    Column(
                        modifier = Modifier
                            .fillMaxWidth()
                            .padding(horizontal = DSSpacing.md, vertical = DSSpacing.sm),
                        verticalArrangement = Arrangement.spacedBy(DSSpacing.xs),
                    ) {
                        Text(
                            text = settingsPath.replace(homeDir, "~"),
                            style = DSFont.monoPath,
                            color = if (configExists) LocalDSColors.current.textPrimary else LocalDSColors.current.textMuted,
                            maxLines = 1,
                        )
                        if (!configExists) {
                            Text(
                                text = "not found — run `gemini` to auto-create on first launch",
                                style = DSFont.sidebarItemSmall,
                                color = LocalDSColors.current.textMuted,
                            )
                        } else {
                            Text(
                                text = "Available keys: theme, sandbox, checkpointing, preferredEditor, contextWindowCompression, telemetry, mcpServers, extensions",
                                style = DSFont.sidebarItemSmall,
                                color = LocalDSColors.current.textMuted,
                            )
                        }
                    }
                }
            }
        },
    )
}
